src/Compilers/Shared/BuildServerConnection.cs

```diff
@@ -588,51 +602,61 @@ internal static string GetClientMutexName(string pipeName)
         }
 
         /// <summary>
-        /// Gets the value of the temporary path for the current environment assuming the working directory
-        /// is <paramref name="workingDir"/>.  This function must emulate <see cref="Path.GetTempPath"/> as 
-        /// closely as possible.
+        /// Gets the value of the temporary path for the provided environment settings. This behavior
+        /// is OS specific: 
+        ///   - On Windows it seeks to emulate Path.GetTempPath as closely as possible with 
+        ///     provided working directory.
+        ///   - On Unix it returns a directory under $XDG_RUNTIME_DIR if it is set using appropriate fall backs
```

Comment:
It looks like the implementation doesn't check this environment variable as described. Instead it's using `$TMPDIR` or `/tmp`. Did I misunderstand something?

Reply:
This PR originally started as trying to fix the temp dir which would include this variable but at the moment it does not. Let me fix this.